lib/osf-components/addon/components/file-browser/file-rename-modal/component.ts

```diff
@@ -51,6 +56,7 @@ export default class FileRenameModal extends Component<Args> {
         } catch(e) {
             this.toast.error(this.intl.t('osf-components.file-browser.file_rename_modal.retry_message'));
         }
+        this.updateOriginalName(newName);
```

Comment:
```suggestion
        this.originalFileName = newName;
```

Reply:
Hey Yuhuai, this was done to prevent potential side effects. I can add type checking to this as well if you like. Do you know if it important to add type to arguments as well as parameters when calling a function?

Reply:
What are some of the potential side effects that are prevented? Doing `this.originalFileName = name` is equivalent to calling a function that does the same thing.

Reply:
The only ways I can think of for a function to modify a property outside of it closure of local scope is: 1. the function modifying something in the global scope or 2. the function modifies the value of parameter passed as reference. In this case, I don't think either applies as both the `updateFileName` task and the `updateOriginalName` function shares the same local scope in the sense that their `this` is the same - the `FileRenameModal` component. Calling `this.originalFileName = newName` from the task is equivalent to calling `this.originalFileName = newName` from the newly added action.

As for the performance concern, I believe tail end optimization gives us boosts when the a function returns a call to another function, in which case the VM does not have to worry about context switching by just jumping to the next function call instead. When there are many recursive function calls, the performance is increased. But I don't think this is the case here: the call to `updateOriginalName` is not a recursive call, nor does the task returns a call to this function. So the VM would still have to save the current context, switch to the context of `updateOriginalName`, then switch back. In fact, not calling the `updateOriginalName` would probably be a slight performance boost.

When you say `the original file name nore properly updating after a save`, I am not sure whether you mean `this.originalFileName` not updating after a save, or the name of the `FileModel` is not updated after a save. The former would not happen regardless of whether we are resetting `this.originalFileName` in the task or in the action. The latter occurs because of something else (namely the fact that we are not reloading the `FileModel` from the API after a save) that will not be fixed by doing this.